build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -16,22 +16,73 @@
 'use strict';
 
 const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs');
+const globby = require('globby');
 const karmaConfig = require('../../test-configs/karma.conf');
+const path = require('path');
+const tempy = require('tempy');
 const testConfig = require('../../test-configs/config');
 const {
   createCtrlcHandler,
   exitCtrlcHandler,
 } = require('../../common/ctrlcHandler');
 const {app} = require('../../server/test-server');
 const {createKarmaServer, getAdTypes} = require('./helpers');
+const {dotWrappingWidth} = require('../../common/logging');
+const {getEsbuildBabelPlugin} = require('../helpers');
 const {getFilesFromArgv} = require('../../common/utils');
-const {getPersistentBrowserifyCache} = require('../../common/browserify-cache');
-const {green, yellow, cyan, red} = require('kleur/colors');
 const {isCiBuild} = require('../../common/ci');
 const {log} = require('../../common/logging');
-const {reportTestStarted} = require('.././report-test-status');
+const {reportTestStarted} = require('../report-test-status');
+const {SERVER_TRANSFORM_PATH} = require('../../server/typescript-compile');
 const {startServer, stopServer} = require('../serve');
 const {unitTestsToRun} = require('./helpers-unit');
+const {yellow, red} = require('kleur/colors');
+
+/**
+ * Used to print dots during esbuild + babel transforms
+ */
+let wrapCounter = 0;
+
+/**
+ * Used to lazy-require the HTML transformer function after the server is built.
+ */
+let transform;
+
+/**
+ * Updates the set of preprocessors to run on HTML and JS files before testing.
+ * Notes:
+ * - The HTML transform is lazy-required because the server is built at startup.
+ * - We must use babel on windows until esbuild can natively downconvert to ES5.
+ * @param {!RuntimeTestConfig} config
+ */
+function updatePreprocessors(config) {
+  const createHtmlTransformer = function () {
+    return function (content, file, done) {
+      if (!transform) {
+        const outputDir = `../../../${SERVER_TRANSFORM_PATH}/dist/transform`;
+        transform = require(outputDir).transformSync;
+      }
+      done(transform(content));
+    };
+  };
+  createHtmlTransformer.$inject = [];
+  config.plugins.push({
+    'preprocessor:htmlTransformer': ['factory', createHtmlTransformer],
+  });
+  const htmlPreprocessors = ['htmlTransformer', 'html2js'];
+  const {fixturesPath} = testConfig;
+  config.preprocessors[fixturesPath] = htmlPreprocessors;
+
+  const jsPreprocessors = ['esbuild'];
+  // TODO(rsimha): Remove this block after evanw/esbuild#297 is fixed.
+  if (argv.ie) {
+    jsPreprocessors.push('babel');
+    config.babelPreprocessor = {options: {caller: {name: 'ie-test'}}};
+  }
+  const unifiedJsFile = [...config.files].pop(); // Inserted by updateFiles()
```

Comment:
Could we make it a global again? The order of the list should be changeable without being aware of this implicit dependency.

Reply:
Good point. Done.